pandas/core/indexes/base.py

```diff
@@ -907,16 +907,26 @@ def astype(self, dtype, copy=True):
         """
 
     @Appender(_index_shared_docs["take"] % _index_doc_kwargs)
-    def take(self, indices, axis=0, allow_fill=True, fill_value=None, **kwargs):
+    def take(
+        self,
+        indices: Union[ArrayLike, Sequence[int]],
```

Comment:
is ArrayLike really needed here?  it should also be a Sequence[int]

Reply:
see #28770 for issue about EA, but applicable to Series, Index and np.ndarray also

```
from typing import Sequence, cast
import numpy as np
import pandas as pd
from pandas.core.arrays import ExtensionArray

df = pd.DataFrame({"a": [1, 2, 3]})
idx = df.index
arr: np.ndarray = cast(np.ndarray, idx._values)
ea: ExtensionArray = pd.array(arr, dtype="Int64")
ser = pd.Series(arr)
reveal_type(idx)
reveal_type(arr)
reveal_type(ea)
reveal_type(ser)


def func(arr: Sequence):
    pass


func(idx)
func(arr)
func(ea)
func(ser)
```
```
/home/simon/t.py:11: note: Revealed type is 'pandas.core.indexes.base.Index'
/home/simon/t.py:12: note: Revealed type is 'numpy.ndarray'
/home/simon/t.py:13: note: Revealed type is 'pandas.core.arrays.base.ExtensionArray'
/home/simon/t.py:14: note: Revealed type is 'pandas.core.series.Series'
/home/simon/t.py:21: error: Argument 1 to "func" has incompatible type "Index"; expected "Sequence[Any]"  [arg-type]
/home/simon/t.py:22: error: Argument 1 to "func" has incompatible type "ndarray"; expected "Sequence[Any]"  [arg-type]
/home/simon/t.py:23: error: Argument 1 to "func" has incompatible type "ExtensionArray"; expected "Sequence[Any]"  [arg-type]
/home/simon/t.py:24: error: Argument 1 to "func" has incompatible type "Series"; expected "Sequence[Any]"  [arg-type]
```